table/arrow_utils.go

```diff
@@ -908,6 +915,260 @@ func must[T any](v T, err error) T {
 	return v
 }
 
+func primitiveToPhysicalType(typ iceberg.Type) string {
+	switch typ.(type) {
+	case iceberg.BooleanType:
+		return "BOOLEAN"
+	case iceberg.Int32Type:
+		return "INT32"
+	case iceberg.Int64Type:
+		return "INT64"
+	case iceberg.Float32Type:
+		return "FLOAT"
+	case iceberg.Float64Type:
+		return "DOUBLE"
+	case iceberg.DateType:
+		return "INT32"
+	case iceberg.TimeType:
+		return "INT64"
+	case iceberg.TimestampType:
+		return "INT64"
+	case iceberg.TimestampTzType:
+		return "INT64"
+	case iceberg.StringType:
+		return "BYTE_ARRAY"
+	case iceberg.UUIDType:
+		return "FIXED_LEN_BYTE_ARRAY"
+	case iceberg.FixedType:
+		return "FIXED_LEN_BYTE_ARRAY"
+	case iceberg.BinaryType:
+		return "BYTE_ARRAY"
+	case iceberg.DecimalType:
+		return "FIXED_LEN_BYTE_ARRAY"
```

Comment:
For the decimal, it depends on the length

![image](https://github.com/user-attachments/assets/51c3a4b9-8966-4ad4-bef0-a63b6bb3f5e4)

Reply:
When I implement the writing to Parquet, the go parquet library already handles mapping Decimal values to int32 or int64 when converting the Arrow -> Parquet schemas

Reply:
Yes, that's a bug in PyIceberg too :) Thanks for catching that!